File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.*;
class Solution {
    public String solution(int[] food) {
        String answer = "";
        
        StringBuilder sb = new StringBuilder();
        
        for(int i = 1; i <= food.length - 1; i++){
            int count = food[i];
            while(count > 1){
                sb.append(i);
                count -= 2;
            }
        }

        answer = sb.toString() + "0" + sb.reverse().toString();
        
        
        return answer;
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(int food[], size_t food_len) {
    int len = 0;
    int idx = 0;
    
    for(int i = 1; i < food_len; i++){
        len += (food[i] / 2) * 2;
    }
    
    char* answer = (char*)malloc((len + 1) * sizeof(char));
    
    for(int j = 1; j < food_len; j++){
        for(int l = 0; l < food[j] / 2; l++){
            answer[idx++] = j + '0';
            answer[(len + 1) - idx] = j + '0';
        }
    }
    
    answer[idx] = '0';
    answer[len + 1] = '\0';
    
    return answer;
}